### PR TITLE
Fix mempool tests

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+﻿using System;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: Are all these `usings` used?

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        public async Task<UnspentOutputs> GetUnspentTransactionAsync(uint256 trxid)
+        {
+            CoinViews.FetchCoinsResponse response = null;
+
+            if (this.coinView != null)
+            {
+                response = await this.coinView.FetchCoinsAsync(new[] { trxid }).ConfigureAwait(false);
+            }
+
+            return response?.UnspentOutputs?.SingleOrDefault();
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: The response is never null

Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: And the unspent outputs inside of the response is also never null

Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: And also, you don't need single or default, you can just take the first element

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: yeah correct was a copy past from the version in master that was used before, I will fix.

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs (hunk 1 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: fixed

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolFeature.cs [hunk 3]
```diff
@@ -171,9 +171,9 @@ public static IFullNodeBuilder UseMempool(this IFullNodeBuilder fullNodeBuilder)
                         services.AddSingleton<BlockPolicyEstimator>();
                         services.AddSingleton<IMempoolValidator, MempoolValidator>();
                         services.AddSingleton<MempoolOrphans>();
-                        services.AddSingleton<MempoolManager>();
-                        services.AddSingleton<IPooledTransaction, MempoolManager>();
-                        services.AddSingleton<IPooledGetUnspentTransaction, MempoolManager>();
+                        services.AddSingleton<MempoolManager>()
+                            .AddSingleton<IPooledTransaction, MempoolManager>(provider => provider.GetService<MempoolManager>())
+                            .AddSingleton<IPooledGetUnspentTransaction, MempoolManager>(provider => provider.GetService<MempoolManager>());
```
Comment (majikandy) [inline src/Stratis.Bitcoin.Features.MemoryPool/MempoolFeature.cs]: Is it correct that these ones are chained on to the previous addsingleton and the others here are on their own line of code?

File: src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        public ConsensusQuery(
+            ICoinView coinView,
+            ILoggerFactory loggerFactory)
+        {
+            this.coinView = coinView;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <inheritdoc />
+        public async Task<UnspentOutputs> GetUnspentTransactionAsync(uint256 trxid)
```
Comment (majikandy) [inline src/Stratis.Bitcoin.Features.Consensus/ConsensusQuery.cs]: Parameter could be named trxId or transactionId

**Pull request merged**